woocommerce/compatibility/class-sv-wc-order-compatibility.php

```diff
@@ -522,6 +522,121 @@ public static function get_edit_order_url( \WC_Order $order ) {
 	}
 
 
+	/**
+	 * Gets the order meta according to HPOS availability.
+	 * Uses WC_Order data store object if HPOS is enabled, otherwise it uses the traditional get_post_meta() method.
+	 *
+	 * @since x.y.z
+	 *
+	 * @param int $order_id Required. Order ID.
+	 * @param string $key Required. Meta key.
+     * @param bool $single return first found meta with key, or all with $key.
+	 * 			   Defaults to true if HPOS is enabled, otherwise false.
+     *
+	 * @return mixed
+	 */
+	public static function get_order_meta( int $order_id, string $key, $single = null) {
+
+		$is_hpos_enabled = SV_WC_Plugin_Compatibility::is_hpos_enabled();
+		$value = false;
+
+		// setting $single to false by default if hpos is disabled to match it with get_post_meta method's default value
+		if( is_null($single) ) {
+			$single = $is_hpos_enabled;
+		}
```

Comment:
arguments in functions need spaces in between parenthesis

prefer `null === $var` than using function call `is_null`

Reply:
also, I'm not sure why we are setting `$is_hpos_enabled` to $single, while methods invoking this method can set their preference; this would cause their preference to be overridden

could you explain the reasoning behind this?

it seems to  me we could as well default this to true though, and drop the $single argument since most of the use cases assume true WRT post meta

Reply:
@unfulvio-godaddy yes, I did not set a default value in the parameter because we'd be using this method for both with/without HPOS functionality, so the default value for `$single` in [get_post_meta](https://developer.wordpress.org/reference/functions/get_post_meta/) is `false`, but in `\WC_Order->get_meta()` the default value of `$single` is set to `true`, so I set these default values accordingly to avoid the ambiguity while replacing `get_post_meta()` with this new method in our plugins, so that we can keep the exact same parameters.

Does that makes sense?

Reply:
@unfulvio-godaddy yes makes sense, as we're getting a single value in most of the use-cases so I will just set it to `true` by default in the arguments to avoid any confusions by anyone using this method.

Reply:
sounds good to me, thanks

Reply:
Yeah I'm not certain that I've _ever_ used the non-single version, it never really made much sense to me

Reply:
> prefer `null === $var` than using function call `is_null`

@unfulvio-godaddy also FWIW as @maxrice pointed out to me the other day, WC coding standards have moved away from yoda conditionals: https://developer.woocommerce.com/2022/08/11/goodbye-yoda-conditions/